src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private RandomSize size = new RandomSize();
+
+    private Figure circle = new Circle(size.getSize(), colorSupplier.getRandomColor());
+    private Figure isoscelesTrapezoid = new IsoscelesTrapezoid(size.getSize(),
+            size.getSize(), size.getSize(), colorSupplier.getRandomColor());
+    private Figure rectangle = new Rectangle(size.getSize(),
+            size.getSize(), colorSupplier.getRandomColor());
+    private Figure square = new Circle(size.getSize(),
+            colorSupplier.getRandomColor());
+    private Figure rightTriangle = new RightTriangle(size.getSize(),
+            size.getSize(), colorSupplier.getRandomColor());

```

Comment:
Don't instantiate figures at the class level when they are supposed to be generated randomly. These should be created inside a method, preferably 'getRandomFigure()', to ensure a new random figure is supplied each time the method is called.